common/workunit/workunit.cpp

```diff
@@ -7297,6 +7297,34 @@ extern WORKUNIT_API StringBuffer &getClusterGroupName(StringBuffer &ret, const c
     return ret;
 }
 
+extern WORKUNIT_API ClusterType getClusterTypeByClusterName(const char *cluster)
+{
+    Owned<IEnvironmentFactory> factory = getEnvironmentFactory(true);
+    Owned<IConstEnvironment> env = factory->openEnvironment();
+    Owned<IPropertyTree> root = &env->getPTree();
```

Comment:
I think ok for now, but this may need to be cached to avoid redoing this every time.
The environment.* implementations have a common cache for environment objects and a table could be build for lookup in their.
Not related to this new method - but ot really sure why a lot of the workunit Cluster type functions are in workunit.* at all.

Reply:
Do you want me to add the cache code now?

Reply:
no, open a separate JIRA to look at.

Reply:
https://track.hpccsystems.com/browse/HPCC-20695